xarray/core/merge.py

```diff
@@ -63,6 +63,9 @@ class Context:
     def __init__(self, func):
         self.func = func
```

Comment:
@keewis do you have an idea of what to do for `groupby.mean` and similar calls?

Reply:
it should be possible to pass the actual (bound?) function instead, so you should also be able to access additional data in the `combine_attrs` function. Not sure how to get a nice `repr` for that but I guess that's a minor issue.